src/components/ConnectWallet.tsx

```diff
@@ -18,15 +18,34 @@ const ConnectWallet = ({ currentTheme }: ConnectWalletProps) => {
   const userSession = useAppSelector(selectUsereSessionState);
   const dispatch = useAppDispatch();
 
+  const currentRole = useAppSelector(selectCurrentUserRole);
+  const location = useLocation();
+
+  const controlAccessRoutes = () => {
+    if (location.pathname !== '/') {
+      if (location.pathname.substring(1).toLowerCase() !== currentRole.toLowerCase()) {
+        console.log('Seems like you got lost, click here to go back to the main page');
+      }
+    }
+  };
+
+  useEffect(() => {
+    controlAccessRoutes();
+  }, [location]);
+
   const disconnect = () => {
     dispatch(disconnectAction());
   };
 
-  const authenticate = () => {
+  const authenticate = async () => {
     dispatch(connectAction());
   };
 
   if (userSession.isUserSignedIn()) {
+    if (currentRole === 'Viewer') {
```

Comment:
Please add another role "None" which is going to be the initial state and add it here.